Enum documentation: Reformulate the returned element for min/max/max_by

PragTob opened the pull request:
The sentence seemed a bit strange, wanted to make it more understandable.
Also wanted to convey that the "compared equal" part only applies/matters to the smallest/biggest element.

Still not sure, if this is really good. Might want to check with a native speaker, to me it read a bit strange so tried to improve upon it.

Also included a second commit, adding the inline doc `iex` for min/max/min_max_by to show the mentioned behavior that the first biggest/smallest element is returned.

2 commits so if wanted only one of them could be taken. Also happy to squash or let github squash it.

lexmag commented on lib/elixir/lib/enum.ex:
```diff
@@ -1181,7 +1181,7 @@ defmodule Enum do
   to Erlang's term ordering.
 
-  If more than one elements compare equal, the first one that was found
+  If multiple elements are considered the biggest, the first one that was found
```
I think `biggest` reads strange and it isn't always true, maybe say `maximal`.
Same applies to `smallest` vs. `minimal`.

PragTob replied:
Thanks for the feedback!

I tried to stay in the language of the documentation, which also uses biggest/smallest - if I'd make that change I would also change that. I'd like the change but wasn't yet ready to make it :)

whatyouhide commented:
Btw, I'm very much :+1: on adding the doctests :)

josevalim commented:
Feel free to merge it Andrea if you have reviewed it. :)


-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

PragTob commented:
I **love** doctests, one of the not enough promoted features of Elixir imo :)

Not sure if the biggest/smalelst --> maximum/minimum change is wanted - let me know what I can do to help this move forward :)

whatyouhide commented:
I'll talk about this with lexmag and get back to you in a couple hours. :)

On Monday, 16 May 2016, Tobias Pfeiffer <notifications@github.com> wrote:

> I *love* doctests, one of the not enough promoted features of Elixir imo
> :)
>
> Not sure if the biggest/smalelst --> maximum/minimum change is wanted -
> let me know what I can do to help this move forward :)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4624#issuecomment-219397415>
>


-- 

Andrea Leopardi
an.leopardi@gmail.com

PragTob commented:
No hurry - all in its own time and pace :) + thanks :+1:

whatyouhide commented:
@ProgTob hey, can you follow @lexmag's suggestion and change "biggest" to "maximal" (same for "minimal" :)), and change this in other places in the docs for these functions as well?

PragTob commented:
@whatyouhide done :)

I changed it up a bit `the maximal of the elements` read strange so I changed it to `the maximal element of the elements` and similarly `considered the maximal` seemed strange so shortened down to `considered maximal` :)

whatyouhide commented on lib/elixir/lib/enum.ex:
```diff
@@ -1178,10 +1178,10 @@ defmodule Enum do
 
   @doc """
-  Returns the biggest of the elements in the enumerable according
+  Returns the maximal element of the elements in the enumerable according
```
This sounds really weird to me 😕 What if we go with "Returns the maximal element in the enumerable"?

PragTob replied:
will do and squash into the last commit

whatyouhide replied:
Awesome, thanks!

PragTob replied:
updated :)

whatyouhide commented:
Thank you @PragTob, great work! 😃 💟

PragTob commented:
Thank you for the good discussion :+1: btw. your talk at ElixirConf.EU was pretty cool and engaging :)

whatyouhide commented:
@PragTob thanks, that's great to hear! :)
